code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm

```diff
@@ -20,6 +21,53 @@
 	else
 		adjustBruteLoss(-20)
 
+/mob/living/simple_animal/demon/shadow/ClickOn(atom/A)
+	if(!ishuman(A))
+		return ..()
+	var/mob/living/carbon/human/target = A
+	if(!in_range(src, target) || target.stat != DEAD)
+		return ..()
+
+	if(isLivingSSD(target) && client.send_ssd_warning(target)) //Similar to revenants, only wrap SSD targets if you've accepted the SSD warning
+		return
+
+	if(wrapping)
+		to_chat(src, "<span class='notice'>We are already wrapping something.</span>")
+		return
+
+	visible_message("<span class='danger'>[src] begins wrapping [target] in shadowy threads.</span>")
+	wrapping = TRUE
+	if(!do_after(src, 4 SECONDS, FALSE, target = target))
+		wrapping = FALSE
+		return
+
+	target.visible_message("<span class='warning'><b>[src] envelops [target] into an ethereal cocoon, and darkness begins to creep from it.</b></span>")
+	var/obj/structure/shadowcocoon/C = new(get_turf(target))
+	target.extinguish_light() // may as well be safe
+	target.forceMove(C)
+	wrapping = FALSE
+
+/obj/structure/shadowcocoon
+	name = "shadowy cocoon"
+	desc = "Something wrapped in what seems to be manifested darkness. Its surface distorts unnaturally, and it emanates deep shadows."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "shadowcocoon"
+	light_power = -4
+	light_range = 6
+	max_integrity = 100
+	light_color = "#AAD84B"
+	anchored = TRUE
+
+/obj/structure/shadowcocoon/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	if(damage_type != BURN) //I unashamedly stole this from spider cocoon code
+		return
+	playsound(loc, 'sound/items/welder.ogg', 100, TRUE)
+
+/obj/structure/shadowcocoon/obj_destruction()
+	visible_message("<span class='danger'>[src] splits open, and the shadows dancing around it fade.</span>")
+	for(var/atom/movable/A in contents)
+		A.forceMove(loc)
```

Comment:
This should probably remain in the `Destroy` as it was before. In 9/10 cases it shouldn't matter. But just for the other 10% it is important to properly clean up the contents.

Reply:
![image](https://user-images.githubusercontent.com/71326864/227041617-0ffe4e76-948c-423a-811a-bd2011965b99.png)